src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,250 @@
+                addDeadline(line);
+                saveFile();
+            } else if (line.startsWith("event")) {
+                addEvent(line);
+                saveFile();
+            } else if (line.startsWith("done")) {
+                markDone(line);
+                saveFile();
+            } else if (line.startsWith("delete")) {
+                deleteTask(line);
+                saveFile();
+            } else {
+                System.out.println("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+            System.out.println("---------------------------------------------------------");
+            line = in.nextLine();
+        }
+        bidGoodbye();
+        saveFile();
+    }
+
+    public static void loadFile() throws IOException{
+        File fileDirectory = new File(dirPath.toString());
+
+        if (!fileDirectory.exists()) {
+            fileDirectory.mkdir();
+        }
+
+        File dataFile = new File(filePath.toString());
+        dataFile.createNewFile();
+        Scanner scanner = new Scanner(dataFile);
+
+        while (scanner.hasNext()) {
+            String line = scanner.nextLine();
+            String type = line.substring(0, 1);
+            String info = line.substring(8);
+            int dateIndex = info.indexOf('|');
+            switch (type) {
+            case "T":
+                tasks.add(new Todo(info));
+                if (line.charAt(4) == 'Y') {
+                    tasks.get(tasks.size() - 1).setAsDone();
+                }
+                break;
+            case "D":
+                tasks.add(new Deadline(info.substring(0, dateIndex - 1), info.substring(dateIndex + 2)));
+                if (line.charAt(4) == 'Y') {
+                    tasks.get(tasks.size() - 1).setAsDone();
+                }
+                break;
+            case "E":
+                tasks.add(new Event(info.substring(0, dateIndex - 1), info.substring(dateIndex + 2)));
+                if (line.charAt(4) == 'Y') {
+                    tasks.get(tasks.size() - 1).setAsDone();
+                }
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    public static void saveFile() {
+        try {
+            FileWriter writer = new FileWriter(filePath.toString());
+            for (Task task : tasks) {
+                if (task instanceof Todo) {
+                    writer.write("T | " + task.getStatusIcon() + " | " + task.getDescription() + System.lineSeparator());
+                } else if (task instanceof Deadline) {
+                    writer.write("D | " + task.getStatusIcon() + " | " + task.getDescription() + " | " + ((Deadline) task).getBy() + System.lineSeparator());
+                } else if (task instanceof Event) {
+                    writer.write("E | " + task.getStatusIcon() + " | " + task.getDescription() + " | " + ((Event) task).getAt() + System.lineSeparator());
+                } else {
```

Comment:
write.write three times here, maybe we should consider to improve the code abstraction and place it in a single method?